server/src/modules/maps/maps.service.ts

```diff
@@ -291,6 +302,21 @@ export class MapsService {
         }
     }
 
+    async delete(mapID: number): Promise<void> {
+        const map = await this.mapRepo.get(mapID);
+        if (!map) throw new NotFoundException('No map found');
+
+        // Deleting all stored map images
+        const imageDtos = await this.getImages(mapID);
+        const imageFileDeletes = imageDtos.map((dto) => this.deleteStoredMapImage(dto.id));
+        await Promise.all(imageFileDeletes);
```

Comment:
Kinda pointless assignment here, I would just do `await Promise.all(imageDtos.map((dto) => this.deleteStoredMapImage(dto.id)));`.